src/format_rules.ml

```diff
@@ -0,0 +1,81 @@
+open Import
+
+let flag_of_kind : Ml_kind.t -> _ =
+  function
+  | Impl -> "--impl"
+  | Intf -> "--intf"
+
+let config_includes (config : Dune_file.Auto_format.t) s =
+  match config.enabled_for with
+  | Default -> true
+  | Only set -> List.mem s ~set
+
+let add_diff sctx loc ~dir input output =
+  let module SC = Super_context in
+  let open Build.O in
+  let action = Action.diff input output in
+  let alias_conf_name = "fmt" in
+  let stamp = ("format-alias", action) in
```

Comment:
Instead of the action, we can just use `input` here, it'll be quicker to hash. We just need something that's unique.